arch/platform/cooja/dev/cooja-radio.c

```diff
@@ -233,24 +223,47 @@ radio_send(const void *payload, unsigned short payload_len)
     return RADIO_TX_ERR;
   }
 
-  /* Transmit on CCA */
-#if COOJA_TRANSMIT_ON_CCA
-  if(send_on_cca && !channel_clear()) {
-    return RADIO_TX_COLLISION;
+  if(radio_was_on) {
+    ENERGEST_SWITCH(ENERGEST_TYPE_LISTEN, ENERGEST_TYPE_TRANSMIT);
+  } else {
+    /* Turn on radio temporarily */
+    simRadioHWOn = 1;
+    ENERGEST_ON(ENERGEST_TYPE_TRANSMIT);
   }
-#endif /* COOJA_TRANSMIT_ON_CCA */
 
-  /* Copy packet data to temporary storage */
-  memcpy(simOutDataBuffer, payload, payload_len);
-  simOutSize = payload_len;
-
-  /* Transmit */
-  while(simOutSize > 0) {
+#if COOJA_SIMULATE_TURNAROUND
```

Comment:
I know this block is placed at the beginning of this function in the original code. But, now, it seems this block should be under the following `else` branch (L247). `ENERGEST_SWITCH(ENERGEST_TYPE_LISTEN, ENERGEST_TYPE_TRANSMIT)` will follow this block.

Reply:
I see your point that the Cooja radio operation energy consumption simulation should be improved. At least I think I do have an idea how to make it more realistic, I'll give it a shot.